File: js/src/formulas/randomVector.ts
export default function randomVector (len: number, norm = 1) {
  const x = []

  let currentNorm = 0

  for (let i = 0; i < len; i++) {
    x.push(Math.random())
    currentNorm += Math.abs(x[i]) ** 2
  }

  currentNorm = currentNorm ** 0.5

  for (let i = 0; i < len; i++) {
    x[i] *= norm / currentNorm
  }

  return x
}

File: cs/src/formulas/randomVector.cs
using System;

public static class RandomVector
{
    public static double[] Generate(int length, double norm = 1)
    {
        var rand = new Random();
        double[] x = new double[length];
        double currentNorm = 0;
        for (int i = 0; i < length; i++)
        {
            x[i] = rand.NextDouble();
            currentNorm += Math.Abs(x[i]) * Math.Abs(x[i]);
        }
        currentNorm = Math.Sqrt(currentNorm);
        for (int i = 0; i < length; i++)
            x[i] *= norm / currentNorm;
        return x;
    }
}
